data/TaskDAO.js

```diff
@@ -0,0 +1,70 @@
+const Task = require('./Task');
+// In-memory implementation
+class TaskDAO {
+    constructor() {
+        this.tasks = [];
+        this.maxID = 0;
+    }
+
+    find(criteria) {
+        if(!criteria || Object.keys(criteria).length === 0) return this.tasks;
```

Comment:
It's a good practice to either use the brackets or not. If you have to use them for the else, then you should use them for the if. Both of these are one liners though so you could do without them if you want to go that way.